packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -3072,7 +3065,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           state,
         );
 
-        if (!arrow.error && !arrow.aborted) return arrow.node;
+        if (!arrow.error && !arrow.aborted && arrow.node) return arrow.node;
```

Comment:
Do we just need an `invariant` here, or is it an actual runtime change?

Reply:
All these checks that I add (in ts and flow file) came from [this](https://github.com/babel/babel/pull/13419#discussion_r652661846)

But during review correction I only updated what you point, I should have pass on all the code to add an invariant comment on simillary cases.

Sorry for that